resolwe/flow/models/data.py

```diff
@@ -726,8 +721,20 @@ def get_runtime_path(self, filename=None):
             prefix=settings.FLOW_EXECUTOR["RUNTIME_DIR"], filename=filename
         )
 
-    def validate_change_collection(self):
+    def move_to_collection(self, destination_collection):
+        """Move data object to collection."""
+        self.validate_change_collection(destination_collection)
+        self.collection = destination_collection
+        if destination_collection:
+            self.tags = destination_collection.tags
+            copy_permissions(destination_collection, self)
+        self.save()
+
+    def validate_change_collection(self, destination_collection):
         """Raise validation error if data object cannot change collection."""
+        if self.collection == destination_collection:
```

Comment:
This should not be necessary as the method should be triggered only for an actual change.